tests/solidity/test_open_short.py

```diff
@@ -0,0 +1,205 @@
+            0,
+            msg=f"{self.hyperdrive.market_state.long_average_maturity_time=} is not correct",
+        )
+        self.assertEqual(
+            self.hyperdrive.market_state.long_base_volume,
+            0,
+            msg=f"{self.hyperdrive.market_state.long_base_volume=} is not correct",
+        )
+        # TODO: once we add checkpointing we will need to switch to this
+        # self.hyperdrive.market_state.long_base_volume_checkpoints(checkpoint_time),
+        # checkpoint_time = maturity_time - self.position_duration
+        self.assertEqual(
+            self.hyperdrive.market_state.long_base_volume,
+            0,
+        )
+        self.assertEqual(
+            self.hyperdrive.market_state.shorts_outstanding,
+            market_state_before.shorts_outstanding + unsigned_bond_amount,
+            msg=f"{self.hyperdrive.market_state.shorts_outstanding=} is not correct",
+        )
+        self.assertEqual(
+            self.hyperdrive.market_state.short_average_maturity_time,
+            maturity_time,
+            msg=f"{self.hyperdrive.market_state.short_average_maturity_time=} is not correct",
+        )
+        self.assertEqual(
+            self.hyperdrive.market_state.short_base_volume,
+            abs(base_amount),
+            msg=f"{self.hyperdrive.market_state.short_base_volume=} is not correct",
+        )
+        # TODO: once we add checkpointing we will need to switch to this
+        # self.hyperdrive.market_state.short_base_volume_checkpoints(checkpoint_time),
+        self.assertEqual(
+            self.hyperdrive.market_state.short_base_volume,
+            abs(base_amount),
+            msg=f"{self.hyperdrive.market_state.short_base_volume=} is not correct",
+        )
+
+    def test_open_short_failure_zero_amount(self):
+        """shorting bonds with zero base fails"""
+        with self.assertRaises(AssertionError):
+            self.hyperdrive.open_short(self.bob.wallet, 0)
+
+    def test_open_short_failure_extreme_amount(self):
+        """shorting more bonds than there is base in the market fails"""
+        # TODO: Shouldn't this be a function of the contribution amount?
+        # The max amount of base does not equal the amount of bonds, it is the result of base_pm.get_max_long
+        bond_amount = self.hyperdrive.market_state.bond_reserves * 2
+        with self.assertRaises(decimal.InvalidOperation):
+            self.hyperdrive.open_short(self.bob.wallet, bond_amount)
+
+    def test_open_short(self):
+        """Open a short & check that accounting is done correctly"""
+        bond_amount = 10
+        self.bob.budget = bond_amount
+        self.bob.wallet.balance = types.Quantity(amount=bond_amount, unit=types.TokenType.BASE)
+        market_state_before = self.hyperdrive.market_state.copy()
+        apr_before = self.hyperdrive.fixed_apr
+        market_deltas, _ = self.hyperdrive.open_short(self.bob.wallet, bond_amount)
+        base_amount = market_deltas.d_base_asset
```

Comment:
might as well just pass this into the function instead of making another variable